Include TerminalReporter in reference docs #6649 (#12661)
pytest-dev/pytest: changelog/6649.doc.rst

> Added `TerminalReporter` to `pytest` root namespace and to API reference documentation page.

Reviewer: ```suggestion
Added :class:`pytest.TerminalReporter` to the :ref:`api-reference` documentation page.
```
